### PR TITLE
Refactor clinic page to use form state hook

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
We have a bug where we're not fully resetting the clinic page's schema when the type of care changes. This PR updates the clinic page to use the form state hook, which takes the schema out of Redux and puts it in local component state. This means it gets removed whenever the page changes, so we don't have to keep it up to date when the type of care changes.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#30502

## Testing done
Local and unit testing

## Acceptance criteria
- [ ] Clinic page works

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/new-appointment/components/ClinicChoicePage.unit.spec.jsx [hunk 1]
```diff
@@ -489,6 +489,7 @@ describe('VAOS <ClinicChoicePage>', () => {
     await waitFor(
       () => expect(screen.getByLabelText(/red team/i).checked).to.be.true,
     );
+    userEvent.click(screen.getByRole('button', { name: /Continue/i }));
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/ClinicChoicePage.unit.spec.jsx:492]: We don't save form data until changing the page with the form state hook, so this test needs to have a continue button click.

File: src/applications/vaos/new-appointment/components/ClinicChoicePage/useClinicFormState.jsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+};
+
+export default function useClinicFormState() {
+  const initialData = useSelector(getFormData);
+  const clinics = useSelector(getClinicsForChosenFacility);
+  const pastAppointments = useSelector(selectPastAppointments);
+
+  const formState = useFormState({
+    initialSchema() {
+      let newSchema = initialSchema;
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/ClinicChoicePage/useClinicFormState.jsx:35]: This is largely copied from the reducer code for the clinic page

File: src/applications/vaos/new-appointment/components/ClinicChoicePage/useClinicFormState.jsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+      return newSchema;
+    },
+    uiSchema,
+    initialData,
+  });
+
+  return {
+    ...formState,
+    firstMatchingClinic: clinics?.find(
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/ClinicChoicePage/useClinicFormState.jsx:103]: Included this here so we don't need to pull the whole clinics list in the clinic page component as well as in this hook.

**Review by vbahinwillit — APPROVED** [review]
LGTM!

**Pull request merged**